src/sidebar/components/top-bar.js

```diff
@@ -1,32 +1,174 @@
+
+const useStore = require('../store/use-store');
+const { applyTheme } = require('../util/theme');
 const isThirdPartyService = require('../util/is-third-party-service');
+const { withServices } = require('../util/service-context');
+
+const GroupList = require('./group-list');
+const SearchInput = require('./search-input');
+const StreamSearchInput = require('./stream-search-input');
+const SortMenu = require('./sort-menu');
+const SvgIcon = require('./svg-icon');
+const UserMenu = require('./user-menu');
+
+/**
+ * The toolbar which appears at the top of the sidebar providing actions
+ * to switch groups, view account information, sort/filter annotations etc.
+ */
+function TopBar({
+  auth,
+  isSidebar,
+  onApplyPendingUpdates,
+  onLogin,
+  onLogout,
+  onSharePage,
+  onShowHelpPanel,
+  onSignUp,
+  pendingUpdateCount,
+  settings,
+}) {
+  const useCleanTheme = settings.theme === 'clean';
+  const showSharePageButton = !isThirdPartyService(settings);
+  const loginLinkStyle = applyTheme(['accentColor'], settings);
+
+  const filterQuery = useStore(store => store.filterQuery());
+  const setFilterQuery = useStore(store => store.setFilterQuery);
+
+  const loginControl = (
+    <Fragment>
+      {auth.status === 'unknown' && (
+        <span className="top-bar__login-links">⋯</span>
+      )}
+      {auth.status === 'logged-out' && (
+        <span className="top-bar__login-links">
+          <a href="#" onClick={onSignUp} target="_blank" style={loginLinkStyle}>
+            Sign up
+          </a>{' '}
+          /{' '}
+          <a href="#" onClick={onLogin} style={loginLinkStyle}>
+            Log in
+          </a>
+        </span>
+      )}
+      {auth.status === 'logged-in' && (
+        <UserMenu auth={auth} onLogout={onLogout} />
+      )}
+    </Fragment>
+  );
+
+  return (
+    <div
+      className={classnames('top-bar', useCleanTheme && 'top-bar--theme-clean')}
+    >
+      {/* Single-annotation and stream views. */}
+      {!isSidebar && (
+        <div className="top-bar__inner content">
+          <StreamSearchInput />
+          <div className="top-bar__expander" />
+          <button
+            className="top-bar__btn top-bar__help-btn"
+            onClick={onShowHelpPanel}
+            title="Help"
+            aria-label="Help"
+          >
+            <SvgIcon name="help" className="top-bar__help-icon" />
+          </button>
+          {loginControl}
+        </div>
+      )}
+      {/* Sidebar view */}
+      {isSidebar && (
+        <div className="top-bar__inner content">
+          <GroupList className="GroupList" auth={auth} />
+          <div className="top-bar__expander" />
+          {pendingUpdateCount > 0 && (
+            <a
+              className="top-bar__apply-update-btn"
+              onClick={onApplyPendingUpdates}
+              title={`Show ${pendingUpdateCount} new/updated annotation(s)`}
+            >
+              <SvgIcon className="top-bar__apply-icon" name="refresh" />
```

Comment:
The `a` element has a title, and the content of the link is an icon...but I seem to remember there being textual content in this link in the past. Am I missing something?

Reply:
It has always been just an icon, with a tooltip.